tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAsset.scala

```diff
@@ -0,0 +1,81 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import java.net.URI
+import java.util.UUID
+
+import com.linagora.tmail.james.jmap.publicAsset.ImageContentType.ImageContentType
+import eu.timepit.refined
+import eu.timepit.refined.api.{Refined, Validate}
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.api.model.Size.Size
+import org.apache.james.jmap.core.AccountId
+import org.apache.james.jmap.mail.BlobId
+import org.apache.james.mailbox.model.ContentType
+
+import scala.util.Try
+
+object PublicAssetIdFactory {
+  def generate(): PublicAssetId = PublicAssetId(UUID.randomUUID())
+}
+
+case class PublicAssetId(value: UUID) {
+  def asString(): String = value.toString
+}
+
+object PublicURI {
+  def fromString(value: String): Either[Throwable, PublicURI] = Try(new URI(value))
+    .map(PublicURI.apply)
+    .toEither
+
+  def from(assetId: PublicAssetId, account: AccountId, publicUriPrefix: URI): PublicURI = {
+    val relativeURI = new URI(s"publicAsset/${account.id.value}/${assetId.asString()}")
```

Comment:
Still string interpolation.

I suggest:

```
        URI baseUri = new URI("https://base.com");
        URIBuilder uriBuilder = new URIBuilder(baseUri);
        URI uri = uriBuilder.setPathSegments(ImmutableList.<String>builder()
                .addAll(uriBuilder.getPathSegments())
                .add("accountId")
                .add("id")
                .build())
            .build();
```

Reply:
Where did you got this URIBuilder?
I tried to URIBUilder that can import in project, but it don't have a API like you want

Reply:
`org.apache.http.client.utils`

Reply:
ah `.setPathSegments` api, thank you